backend/CircleForms/Controllers/PostsController.cs

```diff
@@ -20,14 +21,116 @@ public class PostsController : ControllerBase
+    public PostsController(IValidator<SubmissionContract> submissionValidator, ILogger<PostsController> logger,
+        IPostRepository postRepository, IMapper mapper)
     {
+        _submissionValidator = submissionValidator;
         _logger = logger;
         _postRepository = postRepository;
         _mapper = mapper;
     }
 
+    private (List<Submission> submissions, string error) ProcessAnswer(Post post,
+        IEnumerable<SubmissionContract> answers)
+    {
+        var questions = post.Questions.ToDictionary(x => x.Id);
+        var answersDictionary = answers.ToDictionary(x => x.QuestionId);
+
+        //If any required fields doesn't filled
+        if (post.Questions.Where(question => !question.Optional)
+            .Any(question => !answersDictionary.ContainsKey(question.Id)))
+        {
+            return (null, "One or more required fields doesn't filled");
+        }
+
+        var resultingAnswers = new List<Submission>();
+        foreach (var (key, value) in answersDictionary)
+        {
+            //If question with id doesn't exist or answer was not provided
+            if (!questions.TryGetValue(key, out var question))
+            {
+                return (null, $"Question with id {key} does not exist");
+            }
+
+            if (question.QuestionType == QuestionType.Choice)
+            {
+                var choice = int.Parse(value.Answer);
+
+                if (choice >= question.QuestionInfo.Count)
+                {
+                    return (null, $"Choice for {key} is not in the range of choice ids");
+                }
+            }
+
+            var answer = _mapper.Map<Submission>(value);
+
+            resultingAnswers.Add(answer);
+        }
+
+        return (resultingAnswers, null);
+    }
+
+    /// <summary>
+    ///     Add answer to a question. (Requires auth)
+    /// </summary>
+    [Authorize]
+    [HttpPost("/posts/{id}/answer")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status409Conflict)]
+    public async Task<IActionResult> Answer(string id, [FromBody] List<SubmissionContract> answerContracts)
+    {
+        var claim = HttpContext.User.Identity?.Name;
+        if (string.IsNullOrEmpty(claim) || !long.TryParse(claim, out var userId))
+        {
+            _logger.LogWarning("User had an invalid name claim on answer: {Claim}", claim);
+
+            return Unauthorized();
+        }
+
+        var validationTasks = answerContracts.Select(x => _submissionValidator.ValidateAsync(x));
+        var results = await Task.WhenAll(validationTasks);
+        foreach (var validationResult in results)
+        {
+            if (!validationResult.IsValid)
+            {
+                return BadRequest(new {errors = validationResult.Errors.Select(x => x.ToString())});
```

Comment:
This is not a correct usage of `BadRequest`
![image](https://user-images.githubusercontent.com/8269193/151979428-e29990ce-0351-4820-acbf-e8bddd372894.png)

You should instead use `BadRequest(validationResult.Errors.Select(x => x.ToString());`

Reply:
Nope